experimental/packages/opentelemetry-instrumentation/src/platform/node/instrumentation.ts

```diff
@@ -85,12 +90,13 @@ export abstract class InstrumentationBase<T = any>
       return exports;
     }
 
-    const version = this._extractPackageVersion(baseDir);
-    module.moduleVersion = version;
-    if (module.name === name) {
+    const pkg = this._extractPackage(baseDir);
+    module.moduleVersion = pkg.version;
+    // if the targeted module is an esm, the name will be the name of its entrypoint
+    if (module.name === name || (pkg.main && path.normalize(`${pkg.name}/${pkg.main}`) === name)) {
```

Comment:
> the name will be the name of its entrypoint

That's a convention of IITM?

Reply:
I believe its a convention of esm module loader